data-serving/scripts/convert-data/converters.py

```diff
@@ -462,28 +462,26 @@ def convert_travel_history(id: str, dates: str, location: str) -> Dict[
           'dateRange': {...}
         }
     '''
-    travel_history = {}
-
+    location_list = None
     try:
-        parsed_location = parse_location(location)
-        if parsed_location:
-            travel_history['location'] = parsed_location
+        location_list = parse_location_list(location)
     except (LookupError, ValueError) as e:
         warn(id, 'travelHistory.location', location, e)
 
-    # It's invalid to have dates without a location; only convert the rest of
-    # the field (i.e. the dates) if location was successfully parsed.
-    if not travel_history.get('location'):
-        return None
-
+    date_range = None
     try:
         date_range = convert_date_range(dates)
-        if date_range:
-            travel_history['dateRange'] = date_range
-    except (TypeError, ValueError) as e:
-        warn(id, 'travelHistory.dateRange', dates, e)
+    except (ValueError) as e:
+        warn(id, 'travelHistory.dateRange', location, e)
+
+    if not location_list and not date_range:
+        return None
+    if not location_list:
+        return [{'dateRange': date_range}]
+    if not date_range:
+        return [{'location': l} for l in location_list if l]
 
-    return [travel_history] if travel_history else None
+    return [{'dateRange': date_range, 'location': l} for l in location_list if l]
```

Comment:
Is the idea behind including date range at this level that it'd be useful to have for each location, but for the existing data, travel history is guaranteed to have either 0 or 1 date range? I agree with that reasoning -- but making sure I grok'd correctly.

Reply:
exactly! i'll leave a comment